### PR TITLE
Update file structure

### PR DESCRIPTION
**Pull request by richmccartney**
#### What's this PR do?

Updates all file structure with in the project. This should copy the same style and naming as per the vanilla framework.
#### Where should the reviewer start?

Pull down the branch and run `npm i` to install all required packages.
#### How should this be manually tested?

Run `gulp test` and `build` to make sure the build works
#### Any background context you want to provide?

_n/a_
#### What are the relevant tickets?
#13
#### Screenshots (if appropriate)

_n/a_
#### Questions:

_n/a_

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -5,6 +5,14 @@ require('es6-promise').polyfill();
...
 var wrench = require('wrench');
 
+/*
+  concatenate all *.js / *.coffee files in the 'gulp' folder
+ */
+wrench.readdirSyncRecursive('./gulp').filter(function(file) {
+  return (/\.(js|coffee)$/i).test(file);
+}).map(function(file) {
+  require('./gulp/' + file);
+});
```
Comment (anthonydillon) [inline gulpfile.js:15]: Is this required?

File: scss/_utilities_spacing.scss [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+////
+
+/// Spacing
+/// @group Spacing
+@mixin maas-spacing {
+  @include spacing($base-spacing-unit);
+}
+
+@mixin spacing($spacing-unit) {
+
```
Comment (anthonydillon) [inline scss/_utilities_spacing.scss]: Remove this white space please.

File: scss/_utilities_spacing.scss [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  $spacing:                       margin padding;
+  $directions:                    top right bottom left;
+  $sizes:                         (tiny: $spacing-unit / 4, small: $spacing-unit / 2, large: $spacing-unit * 2, huge: $spacing-unit * 4);
```
Comment (anthonydillon) [inline scss/_utilities_spacing.scss]: Cant you knock these down onto multiple lines please.

``` scss
$sizes: (
    tiny: $spacing-unit / 4,
    small: $spacing-unit / 2,
    ...
);
```

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Looks good with some very small suggested tweaks.

File: gulpfile.js (hunk 1 above)
Comment (richmccartney) [reply gulpfile.js:15]: Yeah it is.. you can't run the gulp test or build with out it, didn't realise until I set it up.

**Comment by richmccartney** [comment]
@anthonydillon all the tweaks should be in now.

**Review by anthonydillon — APPROVED** [review]

**Comment by anthonydillon** [comment]
Looks great!

**Comment by anthonydillon** [comment]
Code looks good but im getting an error running `gulp build`:

```
events.js:160
      throw er; // Unhandled 'error' event
      ^
Error: scss/build.scss
Error: File to import not found or unreadable: ../node_modules/cloud-vanilla-theme/scss/theme
       Parent style sheet: /home/ant/projects/maas-gui-vanilla-theme/scss/build.scss
        on line 2 of scss/build.scss
>> @import '../node_modules/cloud-vanilla-theme/scss/theme';
```

**Comment by richmccartney** [comment]
@anthonydillon weird, I've removed all mention of cloud vanilla theme from this now. ```rm -rf node_modules``` and then run it again.

**Comment by richmccartney** [comment]
@anthonydillon ah yeah it should now be using the Vanilla Framework and not Cloud Vanilla Theme. It will now spit out variable errors when trying to build which is what I expected.

**Comment by anthonydillon** [comment]
Thanks for fixing that. I'm getting another error now:

```
Error: scss/_settings_defaults.scss
Error: Undefined variable: "$warm-grey".
        on line 27 of scss/_settings_defaults.scss
>> $alt-headline-color:    $warm-grey;
```

**Comment by richmccartney** [comment]
@anthonydillon yeah this is expected now. I will be fixing up everything bit by bit as this is getting merged into a development branch

**Comment by anthonydillon** [comment]
Ah ok. :+1: with broken QA then.

**Pull request merged**